### PR TITLE
bug_fix_2512

### PR DESCRIPTION
**Pull request by SadiHassan**
### Changelog category (leave one):

- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

`between()` is doing excess dots check now.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/queries/0_stateless/02366_kql_operators.sql [hunk 1]
```diff
@@ -23,3 +23,6 @@ TableWithVariousDataTypes | project JoinDate | where JoinDate !between (datetime
 TableWithVariousDataTypes | project Age | where Age between (10 .. 12) or Age > 30;
 TableWithVariousDataTypes | project Age | where Age between (10 .. 12) or Age between (30 .. 50);
 range x from 1 to 100 step 1 | where x between ( 8 + (4 * 2) / 8 + ((5 * 5 - 20) * 2) .. 20);
+range x from 1 to 100 step 1 | where x between ( 8 + (4 * 2) / 8 + ((5 * 5 - 20) * 2) ... 20); -- {clientError 62. DB::Exception: Syntax error, number of dots do not match. (SYNTAX_ERROR)}
+range x from 1 to 100 step 1 | where x between ( 8 + (4 * 2) / 8 + ((5 * 5 - 20) * 2) . 20); -- {clientError 62. DB::Exception: Syntax error, number of dots do not match. (SYNTAX_ERROR)}
+range x from 1 to 100 step 1 | where x between ( 8 + (4 * 2) / 8 + ((5 * 5 - 20) * 2)  20); -- {clientError 62. DB::Exception: Syntax error, number of dots do not match. (SYNTAX_ERROR)}
```
Comment (ltrk2) [inline tests/queries/0_stateless/02366_kql_operators.sql:28]: You don't need the entire error message, you can simply go with `-- { clientError SYNTAX_ERROR }`.

**Review by ltrk2 — CHANGES_REQUESTED** [review]

File: tests/queries/0_stateless/02366_kql_operators.sql (hunk 1 above)
Comment (SadiHassan) [reply tests/queries/0_stateless/02366_kql_operators.sql:28]: that's fine. I want to keep like this :)

Comment (ltrk2) [reply tests/queries/0_stateless/02366_kql_operators.sql:28]: Okay, no problem.

Comment (SadiHassan) [reply tests/queries/0_stateless/02366_kql_operators.sql]: Never mind, it really looked messy. code updated :)

**Review by ltrk2 — APPROVED** [review]

**Pull request merged**